pandas/core/tools/numeric.py

```diff
@@ -228,17 +246,28 @@ def to_numeric(
                     if values.dtype == dtype:
                         break
 
-    # GH33013: for IntegerArray & FloatingArray need to reconstruct masked array
-    if mask is not None:
+    # GH33013: for IntegerArray, BooleanArray & FloatingArray need to reconstruct
+    # masked array
+    if (mask is not None or new_mask is not None) and not is_object_dtype(values.dtype):
+        if mask is None:
+            mask = new_mask
```

Comment:
```suggestion
        if mask is None:
            mask = new_mask
        else:
            mask = mask.copy()
```

and then remove the `.copy()` below when constructing the array (right now the mask is always copied, but assuming this newly generated mask doesn't need that)

Reply:
DOne